pkg/hintrunner/zero/utils.go

```diff
@@ -0,0 +1,20 @@
+package zero
+
+import (
+	"math/big"
+
+	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
+)
+
+func AsInt(valueFelt *fp.Element) big.Int {
+	var valueBig big.Int
+	valueFelt.BigInt(&valueBig)
+	boundBig := new(big.Int).Div(fp.Modulus(), big.NewInt(2))
+
+	// val if val < prime // 2 else val - prime
+	if valueBig.Cmp(boundBig) == -1 {
+		return valueBig
+	}
+	negativeValueBig := new(big.Int).Sub(&valueBig, fp.Modulus())
+	return *negativeValueBig
+}
```

Comment:
I also like the idea of passing in a `result *big.Int` argument. So kind of like how the normal `big.Int` api's work. And use the passed `result` to calculate and return the final value. Any thoughts?

Reply:
I think, that this method directly transforms a value from the field arithmetics to the integer. If the value is negative, lets say -1, it gets transformed into PRIME - 1. That is why we subtract the PRIME from the number, to get it's actual value. Which would also mean, that all values >=PRIME // 2 are in fact negative in values, represented by felts, which I'm not sure is correct.

Reply:
I agree with having  `*fp.Element` as argument instead of `*big.Int `, as the idea would be to make the conversion in this method. 

Also, I haven't seen this method used with a different prime, so for now I think we can leave it as it is. Let's leave a comment at the top regarding this.

I would rather for this method to return a pointer to bigInt instead, most of the operations at the end are done using it

Reply:
I feel having a `*fp.Element` is less flexible. If the first thing the function does is converting the argument to `big.int` then might as well make the parent pass it in right? That way the parent still has a reusable element and big.int it can use downstream 🤔